libraries/AC_Avoidance/AP_OAPathPlanner.h

```diff
@@ -88,7 +90,11 @@ class AP_OAPathPlanner {
     bool _thread_created;           // true once background thread has been created
     AP_OABendyRuler *_oabendyruler; // Bendy Ruler algorithm
     AP_OADijkstra *_oadijkstra;     // Dijkstra's algorithm
+    AP_OADatabase _oadatabase;      // Database of dynamic polygon fences
```

Comment:
This isn't being minimized, which means the database is always present in the build, even though users aren't seeing the parameters. Effectively this creates the situation where you have saved very little flash, while still including the feature in a non configurable version.

Reply:
an empty class is being created when minimized
https://github.com/ArduPilot/ardupilot/pull/11671/files#diff-d5888bea59318da2160f3dd6eb3471e1R93

This is done because you can't get a singleton to an unknown class and then even if it's false, try to call isHealthy() on it like what the proximity library does.